lib/whitehall/asset_manager_storage.rb

```diff
@@ -51,17 +54,27 @@ def delete
     end
 
     def url
-      URI.join(Plek.asset_root, Addressable::URI.encode(@legacy_url_path)).to_s
-    end
-
-    def filename
-      ::File.basename(@legacy_url_path)
+      if @model && @model.respond_to?("use_non_legacy_endpoints") && @model.use_non_legacy_endpoints
+        return nil unless @model.assets.any?
+
+        asset_variant = @version ? Asset.variants[@version] : Asset.variants[:original]
+        asset = @model.assets.where(variant: asset_variant).first
```

Comment:
Instead of a database lookup, this just loop through assets and pick the correct variant. That way the test doesn't need to do `model.save!`